src/validation.c

```diff
@@ -415,6 +420,51 @@ validate_sriov_rules(const NetplanParser* npp, NetplanNetDefinition* nd, GError*
     return valid;
 }
 
+gboolean
+adopt_and_validate_vrf_routes(const NetplanParser *npp, GHashTable *netdefs, GError **error)
+{
+    gpointer key, value;
+    GHashTableIter iter;
+
+    g_hash_table_iter_init (&iter, netdefs);
+    while (g_hash_table_iter_next (&iter, &key, &value))
+    {
+        NetplanNetDefinition *nd = value;
+        if (nd->type != NETPLAN_DEF_TYPE_VRF || !nd->routes)
+            continue;
+
+        /* Routes */
+        for (size_t i = 0; i < nd->routes->len; i++) {
+            NetplanIPRoute* r = g_array_index(nd->routes, NetplanIPRoute*, i);
+            if (r->table == nd->vrf_table)
+                continue;
+            else if (r->table != NETPLAN_ROUTE_TABLE_UNSPEC && r->table != nd->vrf_table) {
```

Comment:
question: can't we remove the later clause of the `&&` as already proven by the earlier `if` ?

Reply:
Yes we can!